enginetest/server_engine.go

```diff
@@ -159,7 +163,15 @@ func (s *ServerQueryEngine) QueryWithBindings(ctx *sql.Context, query string, pa
 			if strings.HasSuffix(err.Error(), "empty statement") {
 				return nil, sql.RowsToRowIter(), nil
 			}
-			return nil, nil, err
+			// Note: we cannot access sql_mode when using ServerEngine
+			//  to use ParseWithOptions() method. Replacing double quotes
+			//  because the 'ANSI' mode is not on by default and will not
+			//  be set on the context after SET @@sql_mode = 'ANSI' query.
+			ansiQuery := strings.Replace(query, "\"", "`", -1)
+			parsed, err = sqlparser.Parse(ansiQuery)
+			if err != nil {
+				return nil, nil, err
+			}
```

Comment:
Bummer that we have to parse the statement in order to know which function to use to execute it! Feels like this could probably be simplified somehow, but I think it's okay as is for now.